src/PhpSpec/Matcher/ScalarMatcher.php

```diff
@@ -127,6 +127,10 @@ private function getCheckerName(string $name)
             return false;
         }
 
+        if (PHP_VERSION_ID >= 70400 && $name === 'real') {
```

Comment:
This is a BC break, as specs will suddenly stop passing. I suggest we trigger a deprecation notice instead (we can even do it for all PHP versions) and switch to `is_float` internally (as that's what the alias is internally).

Reply:
Sounds like a great idea !

Reply:
I have just added `trigger_error()` with the deprecation but it is catches by PhpSpec so the specs will still fail